src/app/_models/category_stats.ts

```diff
@@ -0,0 +1,6 @@
+import {Category} from '@app/_models/category';
+
+export interface CategoryStats{
```

Comment:
This model only keeps the categories in a nested format. So it should have a better name (e.g NestedCategories) and should be moved to the category model file.

Reply:
Model has been moved to Category, and renamed to NestedCategories. Rest of code has been refactored to reflect this.